### PR TITLE
Convert Matrix into a function that can filter based on track

### PR DESCRIPTION
**Pull request by maddy531**
This PR converts Matrix into a function that takes in a track.
If no track is provided then the full list of competencies are listed.

Issue: https://github.com/Khan/career-competencies/issues/27

### CODE REVIEW

File: src/data/matrix.ts [hunk 1]
```diff
@@ -19,27 +20,35 @@ export const Matrix: {
...
+};
+
+
+const filterByTrack = (selectedTrack, competency: Competency) => {
+  return !selectedTrack || competency.track === "Core" || selectedTrack === competency.track;
+}
+
+export const Matrix: MatrixType = (selectedTrack) => (
+  {
+    byCompetency: Competencies.filter((competency) => filterByTrack(selectedTrack, competency)).map((competency) => ({
```
Comment (maddy531) [inline src/data/matrix.ts:32]: before going spreading the competencies, filter it based on task first.

File: src/data/matrix.ts [hunk 2]
```diff
@@ -19,27 +20,35 @@ export const Matrix: {
...
+            skill.competency === competency.key &&
+            skill.expectation === expectation.key,
+        ),
+      })),
     })),
-  })),
-};
+    byExpectation: PrimaryExpectations.map((expectation) => ({
+      ...expectation,
+      competencies: Competencies.filter((competency) => filterByTrack(selectedTrack, competency)).map((competency) => ({
```
Comment (maddy531) [inline src/data/matrix.ts:45]: Do the same filtering here

**Comment by nedredmond** [comment]
@maddy531 thanks! Can you try running `bun run fix`?

**Comment by nedredmond** [comment]
Maddy, you can absolutely call `const { user } = useData()` and use `Matrix(user.track).by...` right now. It'll just show us Core since there that is the default for null info. I don't think it needs a follow-up task.

**Pull request merged**